internal/expr/expr_test.go

```diff
@@ -442,6 +464,10 @@ func (s *ExprSuite) TestPrepareErrors(c *C) {
 		query:       "SELECT (&Person.*, &Person.*) FROM t",
 		prepareArgs: []any{Address{}, Person{}},
 		err:         `cannot prepare expression: member "address_id" of type "Person" appears more than once`,
+	}, {
+		query:       "SELECT (&M.id, &M.id) FROM t",
+		prepareArgs: []any{sqlair.M{}},
+		err:         `cannot prepare expression: member "id" of type "M" appears more than once`,
```

Comment:
This is actually added as part of the [bugfix here](https://github.com/canonical/sqlair/pull/98) which is included in this PR. We throw an error here because even though its perfectly possible to select the same column twice, here it being read into the same place twice. This would have no effect so is almost certainly a mistake, hence the error.